### DISCUSSION TITLE
Accessing errno and constants, such as ENOEXEC in JS

### DISCUSSION THREAD

**Discussion by iliakan** [emscripten-core/emscripten#18880] [Q&A] [answered]
Is there a way to set C `errno` from JS? Should I create a special C-function for that `void set_errno(int my_errno)`, or there's a more elegant way?

I see a built-in `__errno_location` function, should I use it? Or maybe stay away from it, cause it's internal?

Also, there're constants, such as ENOEXEC and others, should I export `ERRNO_CODES` to access them in JavaScript (for the purpose of returning to C)?

**Comment by sbc100** [comment]
If the code you are writing is in a library JS file you can use the preprocessor to get access to them like this: `var code = {{{ cDefine('EINVAL') }}}`.

Otherwise you can use the ERRNO_CODES library helper: https://github.com/emscripten-core/emscripten/blob/168dafbfe4521a70096e4aadb3a0beb3813ffb15/src/library.js#L1282

You can force that to be include with `-sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE=$ERRNO_CODES`, and exported using `-sEXPORTED_FUNCTIONS=ERRNO_CODES`  (node the $ in the first list and the lack of $ in the second).    If you want of those then I think you can also do `-sEXPORTED_RUNTIME_METHODS=ERRNO_CODES`...

**Comment by iliakan** [comment, marked as answer]
Thankd.

Solved by `-sEXPORTED_RUNTIME_METHODS=setErrNo,ERRNO_CODES` =)
